daemons/controld/controld_join_dc.c

```diff
@@ -476,25 +467,28 @@ do_dc_join_filter_offer(long long action,
          * element from the join client. The "if" guard is for clarity.
          */
         if (pcmk__xe_is(generation, PCMK__XE_GENERATION_TUPLE)) {
-            for (lpc = 0; cmp == 0 && lpc < PCMK__NELEM(attributes); lpc++) {
+            for (int i = 0; (cmp == 0) && (i < PCMK__NELEM(attributes)); i++) {
                 cmp = compare_int_fields(max_generation_xml, generation,
-                                         attributes[lpc]);
+                                         attributes[i]);
             }
 
         } else {    // Should always be PCMK__XE_GENERATION_TUPLE
             CRM_LOG_ASSERT(false);
         }
```

Comment:
This could also be inverted:

```
if (!pcmk__xe_is(generation, PCMK__XE_GENERATION_TUPLE)) {
    CRM_LOG_ASSERT(false);
}

for (int i = ...) {
}
```

Reply:
It would also be possible to functionize some more of the body of this function, such as the comparison. Then we could `CRM_CHECK()` -> return early, instead of having the if-else-CRM_LOG_ASSERT().

This PR is big enough, but I'll try to remember to do that the next time I'm in this code.